src/main/java/com/puppycrawl/tools/checkstyle/Main.java

```diff
@@ -786,24 +786,32 @@ else if (hasConfigurationFile) {
                     final String msg = "Could not find config XML file '%s'.";
                     result.add(String.format(Locale.ROOT, msg, configurationFile));
                 }
-
-                // validate optional parameters
```

Comment:
why start here and not include the try/catch above? Then the whole new method could be called like `validateCliWithConfigurationFile` or such.

Reply:
config file is not optional, all other parameters are optional.
I think it is reasonable in current form, please review again.

Reply:
I was referring to this is all part of the if.
> else if (hasConfigurationFile) {

Moving the try/catch to the new method would make the code:
````
else if (hasConfigurationFile) {
    result.addAll(validateOptionalCliParameters());
}
````

Reply:
Config is not optional in comparison to MT parameters

Reply:
sounds fine to me. I was only speaking about putting more code into the method.

Reply:
renamed, with more code ... it become hard to make name, .... we will think about it later.